<!-- -->
<launch>
    <arg name="lidar_frame_id" default="os_sensor" />

    <!-- Set localization method: -->
    <arg name="topic_pose_stamped" default="/ground_truth_pose"/>

    <!-- Set args defining tf_transform: base_link to lidar-->
    <arg name="tf_x" default="0.6"/>
    <arg name="tf_y" default="0.0"/>
    <arg name="tf_z" default="2.0"/>
    <arg name="tf_yaw" default="0.0"/>
    <arg name="tf_pitch" default="0.0"/>
    <arg name="tf_roll" default="0.0"/>

    <!-- tf_transform: base_link to lidar -->
    <include file="$(find runtime_manager)/launch_files/setup_tf.launch">
        <arg name="x" value="$(arg tf_x)"/>
        <arg name="y" value="$(arg tf_y)"/>
        <arg name="z" value="$(arg tf_z)"/>
        <arg name="yaw" value="$(arg tf_yaw)"/>
        <arg name="pitch" value="$(arg tf_pitch)"/>
        <arg name="roll" value="$(arg tf_roll)"/>
        <arg name="frame_id" value="base_link"/>
        <arg name="child_frame_id" value="$(arg lidar_frame_id)"/>
        <arg name="period_in_ms" value="10"/>
    </include>

    <!-- Vehicle Model -->
    <include file="$(find vehicle_description)/launch/vehicle_model.launch">
        <arg name="model_path" value="$(find adeye)/../../../../AD-EYE/Data/URDF_Actor_Models/BMW_X5/default.urdf"/>
    </include>

    <!-- Points Downsampler -->
    <node pkg="rostopic" type="rostopic" name="config_voxel_grid_filter_rostopic"
          args="pub -l /config/voxel_grid_filter autoware_config_msgs/ConfigVoxelGridFilter
        '{voxel_leaf_size: 0.2 , measurement_range: 200.0}' "
    />
    <include file="$(find points_downsampler)/launch/points_downsample.launch"/>

    
    <!-- Fake Localizer, publishes the transform from map to base link base on gnss data -->
    <node type="fake_localizer.py"
        pkg="adeye"
        name="ground_truth_localizer"
        args="ground_truth_localizer /map /base_link"/>

    <!-- Pose Connect -->
    <node pkg="topic_tools" type="relay" name="pose_relay" output="log" args="$(arg topic_pose_stamped) /current_pose"/>

</launch>
